### PR TITLE
Add prometheus client to monitor the API

### PR DESCRIPTION
**Pull request by TristanCacqueray**
This change adds a new middleware to serve ghc metrics on the
/metrics endpoint.

### CODE REVIEW

**Comment by TristanCacqueray** [comment]
While this is already working, we should probably add the prometheus service + grafana to the compose and document its usage before merging this PR.

File: nix/default.nix [hunk 1]
```diff
@@ -66,6 +68,68 @@ let
...
+  grafana-home = "/tmp/grafana-home";
+  grafanaPromDS = pkgs.writeTextFile {
+    name = "prometheus.yml";
+    text = ''
+      apiVersion: 1
+      datasources:
+      - name: Prometheus
+        type: prometheus
+        access: direct
+        url: http://localhost:19090
```
Comment (morucci) [inline nix/default.nix:102]: url: http://localhost:${toString prom-port}

File: nix/default.nix [hunk 2]
```diff
@@ -66,6 +68,68 @@ let
...
+    text = ''
+      [security]
+      admin_user = admin
+      admin_password = monocle
+
+      [server]
+      http_port = ${toString grafana-port}
+
+      [auth.google]
+      client_secret = 0ldS3cretKey
```
Comment (morucci) [inline nix/default.nix:129]: Is this needed ?

**Review by morucci — APPROVED** [review]
I think we can merge your change. It seems to work as expected. I'll create an issue to keep track of the remaining tasks.

**Comment by TristanCacqueray** [comment]
Actually the middleware might not be so useful, it would be better to manage the application metrics manually, e.g. using a label for the search endpoint for example.

**Review by morucci — APPROVED** [review]

**Pull request merged**